tagilmo/utils/vereya_wrapper.py

```diff
@@ -490,6 +499,8 @@ def _update_cache(self, method):
                 self.cached_buffer[method] = self.cached[method]
                 self.cached[method] = (v_new, t_new)
```

Comment:
Maybe it will be more safe to put chat messages in a queue and zero it out on read event? This way we will never miss a chat message.

Reply:
@noskill  well, i thought about that. But in case of the list of chat messages, which one we will read? Last one and drop previous ones?

Reply:
> @noskill well, i thought about that. But in case of the list of chat messages, which one we will read? Last one and drop previous ones?

I'd say that the processor side (agent) just pops up messages - as many of them as needed. It can read the last one and discard the rest; or it will need to process all of them.

Reply:
@DaddyWesker The code should return all the messages on read event, not only the last.

Reply:
@noskill  and what if list will contain messages which interrupt each other? For example, "get stone" and next one is "get log"?

Reply:
It's up to the client code to decide what to do.

Reply:
@noskill ok

Reply:
@noskill  @Necr0x0Der  I've changed events to lists. Check if it is what you've expected.